### PR TITLE
txn: bump checksum version

### LINKED ISSUE #57206: bump checksum version to 2 to make it compatible to previous version
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

* deploy cluser v8.3.0 write in some data
* upgrade to v8.4.0
* the checksum may cannot be verified since v8.3.0 use key instead of the handle

### 2. What did you expect to see? (Required)

verification passed

### 3. What did you see instead (Required)

failed since use different element to calculate the cheksum

### 4. What is your TiDB version? (Required)

v8.3.0 bump to v8.4.0

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #57206 

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

```
deploy tidb cluster v8.3.0

create table t(a int primary key, b int);
insert into t values (1, 2);

upgrade tidb to v8.4.0

update t set b = 3 where a = 1;

update t set a = 2 where a = 1;
```

output:
```

[2024/11/07 18:50:10.138 +08:00] [INFO] [mounter.go:636] ["raw bytes checksum matched"] [version=1] [expected=1383743328] [obtained=1383743328]

[2024/11/07 18:55:49.438 +08:00] [INFO] [mounter.go:636] ["raw bytes checksum matched"] [version=1] [expected=1383743328] [obtained=1383743328]
[2024/11/07 18:55:49.438 +08:00] [INFO] [mounter.go:636] ["raw bytes checksum matched"] [version=2] [expected=3631181330] [obtained=3631181330]

[2024/11/07 18:56:07.438 +08:00] [INFO] [mounter.go:636] ["raw bytes checksum matched"] [version=2] [expected=3631181330] [obtained=3631181330]
[2024/11/07 18:56:07.438 +08:00] [INFO] [mounter.go:636] ["raw bytes checksum matched"] [version=2] [expected=1097220008] [obtained=1097220008]

```

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

File: pkg/util/rowcodec/encoder.go [hunk 1]
```diff
@@ -239,7 +239,10 @@ func (NoChecksum) encode(encoder *Encoder, buf []byte) ([]byte, error) {
 	return encoder.toBytes(buf), nil
 }
 
-const checksumVersionRaw byte = 1
+// introduced since v7.1.0
+// bumped to 1 since v8.3.0
+// bumped to 2 since v8.4.0
+const checksumVersionRaw byte = 2
```
Comment (tangenta) [inline pkg/util/rowcodec/encoder.go:245]: Is it better to use three constants instead and describe what's changed in the comment for each of them?

**Review by zyguan — APPROVED** [review]

**Pull request merged**